pkg/authentication/x509/x509_enabler_test.go

```diff
@@ -0,0 +1,104 @@
+package x509
+
+import (
+	"testing"
+
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/automationconfig"
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/util/authtypes"
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/util/constants"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestX509AutomationConfig(t *testing.T) {
+	// Only X509
+	auth := automationconfig.Auth{}
+	opts := authtypes.Options{
+		AuthoritativeSet:  false,
+		KeyFile:           constants.AutomationAgentKeyFilePathInContainer,
+		AuthMechanisms:    []string{constants.X509},
+		AutoAuthMechanism: constants.X509,
+	}
+	err := configureInAutomationConfig(&auth, "keyfilecontents", "CN=my-agent,O=MongoDB", []automationconfig.MongoDBUser{}, opts)
+	assert.NoError(t, err)
+
+	t.Run("Authentication is correctly configured", func(t *testing.T) {
+		assert.Equal(t, "CN=my-agent,O=MongoDB", auth.AutoUser)
+		assert.Equal(t, "keyfilecontents", auth.Key)
+		assert.Equal(t, "", auth.AutoPwd)
+		assert.Equal(t, constants.X509, auth.AutoAuthMechanism)
+		assert.Len(t, auth.DeploymentAuthMechanisms, 1)
+		assert.Len(t, auth.AutoAuthMechanisms, 1)
+		assert.Equal(t, []string{constants.X509}, auth.DeploymentAuthMechanisms)
+		assert.Equal(t, []string{constants.X509}, auth.AutoAuthMechanisms)
+		assert.Equal(t, constants.AutomationAgentKeyFilePathInContainer, auth.KeyFile)
+		assert.Equal(t, constants.AutomationAgentWindowsKeyFilePath, auth.KeyFileWindows)
+	})
+	t.Run("Subsequent configuration doesn't add to deployment auth mechanisms", func(t *testing.T) {
+		err := configureInAutomationConfig(&auth, "keyfilecontents", "CN=my-agent,O=MongoDB", []automationconfig.MongoDBUser{}, opts)
+		assert.NoError(t, err)
+		assert.Equal(t, []string{constants.X509}, auth.DeploymentAuthMechanisms)
+	})
+
+	// X509 and SHA-256
```

Comment:
as you use comments to split them I would suggest to either:
- create a dedicated method for each 
- or group them via {} scopes